src/vm/script.rs

```diff
@@ -1260,14 +1260,276 @@ impl<'a> ScriptExecutor<'a> {
+                    *memory_size -= e1.size();
+                    let e2 = exec_stack.pop().unwrap();
+                    *memory_size -= e2.size();
+
+                    if e1 >= e2 {
+                        self.state = ScriptExecutorState::ContinueLoop;
+                    } else {
+                        self.state = ScriptExecutorState::BreakLoop;
+                    }
+                }
+
+                ScriptEntry::Opcode(OP::ContinueIfLt) => {
+                    if exec_stack.len() < 2 {
+                        self.state = ScriptExecutorState::Error(
+                            ExecutionResult::InvalidArgs,
+                            (i_ptr, func_idx, op.clone(), exec_stack.as_slice()).into(),
+                        );
+                        return;
+                    }
+
+                    let e1 = exec_stack.pop().unwrap();
+                    *memory_size -= e1.size();
+                    let e2 = exec_stack.pop().unwrap();
+                    *memory_size -= e2.size();
+
+                    if e1 < e2 {
+                        self.state = ScriptExecutorState::ContinueLoop;
+                    } else {
+                        self.state = ScriptExecutorState::BreakLoop;
+                    }
+                }
+
+                ScriptEntry::Opcode(OP::ContinueIfGt) => {
+                    if exec_stack.len() < 2 {
+                        self.state = ScriptExecutorState::Error(
+                            ExecutionResult::InvalidArgs,
+                            (i_ptr, func_idx, op.clone(), exec_stack.as_slice()).into(),
+                        );
+                        return;
+                    }
+
+                    let e1 = exec_stack.pop().unwrap();
+                    *memory_size -= e1.size();
+                    let e2 = exec_stack.pop().unwrap();
+                    *memory_size -= e2.size();
+
+                    if e1 > e2 {
+                        self.state = ScriptExecutorState::ContinueLoop;
+                    } else {
+                        self.state = ScriptExecutorState::BreakLoop;
+                    }
+                }
+
                 ScriptEntry::Opcode(OP::End) => {
                     self.state = ScriptExecutorState::EndBlock;
                 }
 
+                ScriptEntry::Opcode(OP::Depth) => {
+                    exec_stack.push(VmTerm::Unsigned16(exec_stack.len() as u16));
+                    *memory_size += 2;
+                }
+
+                ScriptEntry::Opcode(OP::IfDup) => {
+                    let len = exec_stack.len();
+                    if len == 0 {
+                        self.state = ScriptExecutorState::Error(
+                            ExecutionResult::InvalidArgs,
+                            (i_ptr, func_idx, op.clone(), exec_stack.as_slice()).into(),
+                        );
+                        return;
+                    }
+
+                    if !exec_stack[len - 1].clone().equals_0() {
```

Comment:
The clone can be removed when `equals_0` is changed to take an immutable reference